inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -1329,7 +1329,15 @@ void InsertSplitAligningFilterPass::run() {
         for (auto &&splitOutput  : l->outData) {
             auto outputSize = product(++begin(splitOutput->getDims()), end(splitOutput->getDims()));
 
-            if (currentOffset != ALIGN64(currentOffset)) {
+            auto outFunctionalLayers = CNNNetGetAllNextLayersSkipCertain(l, -1, [](CNNLayerPtr next_layer) {
```

Comment:
search of padding can be moved before the loop for split outputs, now it iterates through all outputs for every iteration of this loop